src/Xamarin.Android.Build.Tasks/Tasks/GetAdditionalResourcesFromAssemblies.cs

```diff
@@ -172,7 +173,7 @@ bool ExtractArchive (string url, string file, string contentDir)
 					}
 				}
 				catch (Exception e) {
-					LogCodedError ("XA5209", "Unzipping failed. Please download {0} and extract it to the {1} directory.", url, contentDir);
+					LogCodedError ("XA5209", "Unzipping failed while attempting to extract {0} into {1}{2}", file, contentDir, Path.DirectorySeparatorChar);
```

Comment:
Here are a few things I was considering while working on this "first draft" of the update for this error message:

- The instructions in the original message were incorrect after monodroid/master e8737f15c5e6f71f01f566475865a2cb34bfdfc9.  The output location for unzipping the file by hand was no longer correct.  Additionally, the new logic expects the downloaded `.zip` file to exist in the "zips" folder.  So the user should now complete at least 3 steps for the "fully manual" procedure: download the file, copy it to the MD5 file name in the "zips" folder, and then extract the main archive.

- At the time when this error happens, "some" version of the file does already exist.  The additional "Reason:" information helps clarify how the unzipping failed.  For example, the problem could be a permissions or I/O error (such as unable to write to a full disk).  In those cases, the user should look carefully at the additional "Reason" to see if re-downloading the file is likely to help.

- If this error occurs when unzipping one of the embedded archives rather than the main outer archive, then there is no way to re-download _just_ the embedded archive.  Additional conditional logic and lengthier instructions would be needed to account for that scenario.  (This could happen due to a disk-full error during unzipping or maybe more likely a "path too long" exception on Windows.)

That said, I'd be happy to compromise between the old and the new error message.  In the particular case where the "Reason" is "File ... is not a ZIP archive", the error does most likely indicate an incomplete download (at least for the Xamarin.Android.Support packages).  And that scenario is arguably the most common reason for a failed unzip.  So I'd be happy to either:

(a) Add some conditional logic to check whether the "Reason" is "not a ZIP archive" and include an additional recommendation about re-downloading the file in that case.  It appears that https://github.com/xamarin/xamarin-android/pull/150 now implements that approach [during the download step itself] (https://github.com/xamarin/xamarin-android/pull/150/files#diff-08ec4c94b09aa9ba9ea76feac1f66066R293), so perhaps we don't have to worry about it anymore during the unzipping step.

or

(b) Include the re-downloading recommendation directly in the message with some "conditional wording."  Something like:

> ("Unzipping failed. Please check the \\"Reason:\\" for additional details. If the reason is \\"not a ZIP archive\\", then the file download might have been incomplete, so a good first step would be to try re-downloading {0} by hand and copying it to {1}", url, file)